packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -606,7 +608,30 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
                 uint256(0), // value
                 uint64(SYSTEM_DEPOSIT_GAS_LIMIT), // gasLimit
                 false, // isCreation,
-                abi.encodeCall(IL1Block.setGasPayingToken, (_token, _decimals, _name, _symbol))
+                abi.encodeCall(IL1Block.setConfig, (_type, _value))
+            )
+        );
+    }
+
+    /// @notice Calls the L2ProxyAdmin as the DEPOSITOR_ACCOUNT. This function can be used
+    ///         to upgrade the predeploys on L2. Only callable by the upgrader role on the
+    ///         SuperchainConfig.
+    function upgrade(uint32 _gasLimit, bytes memory _calldata) external {
```

Comment:
@tynes any reason we don't just pass this in as `uint64` instead of casting it below?

Reply:
It could make more sense to use u64 given that the messenger uses u64 for its gas param. Better to be consistent